apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -34,45 +49,136 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
 
+                items = items.filter((item) => organizationsIdsWithAcquiringContext.has(item.organizationId))
 
-                const getPayload = (accountNumber, resident) => ({
-                    dv,
-                    sender,
-                    resident: { connect: { id: resident.id } },
-                    organization: { connect: { id: get(resident, ['organization', 'id']) } },
-                    accountNumber: accountNumber,
+                // Filter by property
+                const organizationsIdsWithProperties = new Set()
+                await loadListByChunks({
+                    context,
+                    list: Property,
+                    chunkSize: 50,
+                    where: {
+                        deletedAt: null,
+                        organization: { id_in: items.map((item) => item.organizationId) },
+                        OR: items.map((item) => ({
+                            OR: [{ address_i: item.address }, { addressKey: item.addressKey }],
+                        })),
+                    },
+                    chunkProcessor: (chunk) => {
+                        chunk.forEach((row) => {
+                            organizationsIdsWithProperties.add(get(row, ['organization', 'id']))
+                        })
+                        return []
+                    },
                 })
 
-                const combinations = _.flatMap(residents, (resident) =>  billingAccounts.map((account) => [resident, account]))
-                const createdServiceConsumers = await Promise.all(combinations.map(([resident, account]) => ServiceConsumer.create(context, getPayload(account.number, resident))))
+                items = items.filter((item) => organizationsIdsWithProperties.has(item.organizationId))
 
-                logger.info({  msg: 'Created ServiceConsumers for input data: ', payload: {
+                const residentsWhere = {
+                    deletedAt: null,
+                    OR: items.map(({ address, addressKey, unitType, unitName }) => ({
+                        AND: [{ OR: [{ address_i: address }, { addressKey }] }, { unitType }, { unitName }],
+                    })),
+                }
+                const residentsFilter = get(filters, 'residentsIds')
+                if (residentsFilter) {
+                    residentsWhere.id_in = residentsFilter
+                }
+                const residents = await Resident.getAll(context, residentsWhere)
+
+                const combinations = flatMap(residents, (resident) => billingAccounts.map((account) => [resident, account]))
+                const createdServiceConsumers = await Promise.all(
+                    combinations.map(([resident, account]) => ServiceConsumer.create(context, {
+                        dv,
+                        sender,
+                        resident: { connect: { id: resident.id } },
+                        organization: { connect: { id: get(resident, ['organization', 'id']) } },
```

Comment:
possible `null`

Reply:
(in case of org deleted or it has not part of our product yet)

Reply:
take org from another place

Reply:
Fixed